public/app/features/panel/PanelRenderer.tsx

```diff
@@ -94,11 +99,13 @@ function useFieldOverrides(
   const series = data?.series;
   const fieldConfigRegistry = plugin?.fieldConfigRegistry;
   const theme = useTheme2();
+  const structureRev = useRef(0);
 
   return useMemo(() => {
     if (!fieldConfigRegistry || !fieldConfig || !data) {
       return;
     }
+    structureRev.current = structureRev.current + 1;
```

Comment:
@mckn @peterholmberg - I've added the structure revision bump over here, whenever the field config changes. It has couple of implications tho. Basically - we do not guarantee referential equality of field configs that are objects( i.e. `hideFrom`) as well as we do not bump the structureRev only in the structure has actually changed. All of the above happens in the PQR that does plenty of stuff to compare the previous vs current data in order to bump or not the structureRev. 

Do you think this simplification here is OK for alerting needs? I would love to talk in the future about how we can re-use PQRs structureRev bump in the PanelRenderer, or alternatively up in the AlertingQueryRunner. This is not a pressing discussion at the moment, but I don't want to unnecessarily compicate the PanelRenderer now with duplicated code of the structureRev calc.

Reply:
Yes it's fine for now. We should revisit implementations of PQRs in the future, maybe post 8.2.